server/export.js

```diff
@@ -14,345 +14,199 @@ limitations under the License.*/
-		var node = map.nodes[index];
-		var cX = Math.floor(node.positionX * availableWidth) + 30;
-		var cY = Math.floor(node.positionY * availableHeight) + 30;
-		calculatedNodes.push({
-			x : cX,
-			y : cY,
-			name : node.name,
-			componentId : node.componentId
-		});
-	}
-	ctx.stroke();
-
-	var recordedConnectionLabels = [];
-	//draw connections
-	for(var index in map.connections){
-		var connection = map.connections[index];
-		var source = connection.pageSourceId;
-		var target = connection.pageTargetId;
-		var scope = connection.scope;
-		var start;
-		var end;
-		for(var nodeIndex in calculatedNodes){
-
-			if(calculatedNodes[nodeIndex].componentId == source){
-				start = calculatedNodes[nodeIndex];
-			}
-			if(calculatedNodes[nodeIndex].componentId == target){
-				end = calculatedNodes[nodeIndex];
+var fs = require('fs');
+var d3 = require('d3'), jsdom = require('jsdom');
+var xmldom = require('xmldom');
+var _ = require('underscore');
+
+
+// TODO: 'Created at www.wardleymaps.com'
+// TODO: map name
+// TODO: map description
+// TODO: scope:actions, scope:jsPlumb_DefaultScope
+
+
+
+var draw = function(res, filename, map){
+	var htmlStub = '<html><head><script></script></head><body><div id="map-container"></div></body></html>';
+	var scripts = ['//cdnjs.cloudflare.com/ajax/libs/d3/3.5.5/d3.js', '//cdnjs.cloudflare.com/ajax/libs/lodash.js/3.2.0/lodash.min.js'];
+	
+	jsdom.env({
+		features : {
+			QuerySelector : true,
+			FetchExternalResources : [ 'script', 'css' ],
+			ProcessExternalResources : [ 'script', 'css' ]
+		},
+		html : htmlStub,
+		scripts : scripts, 
+		done : function(errors, window) {
+			
+			function pick(key) {
+				return function(d) {
+					if(typeof d[key] === 'string' && d[key] === 'true'){
+						return true;
+					}
+					if(typeof d[key] === 'string' && d[key] === 'false'){
+						return false;
+					}
+					return d[key];
+				};
 			}
 
-		}
-		if(scope == "jsPlumb_DefaultScope"){
-			ctx.strokeStyle = 'silver';
-			ctx.moveTo(start.x, start.y);
-			ctx.lineTo(end.x,end.y);
-			ctx.stroke();
-		}
-		if(scope == "Actions"){
-			ctx.strokeStyle = 'green';
-			var lineLength = Math.sqrt((start.x - end.x)*(start.x - end.x) + (start.y - end.y)*(start.y - end.y));
-			var deltaX = 10 * (start.x - end.x) / lineLength;
-			var deltaY = 10 * (start.y - end.y) / lineLength;
-			drawLineArrow(ctx, start.x, start.y, end.x + deltaX, end.y + deltaY);
-			// record a label
-			if(connection.label){
-				var labelX = ( start.x + end.x ) / 2 ;
-				var labelY = ( start.y + end.y ) / 2 ;
-				recordedConnectionLabels.push({label:connection.label,x:labelX, y:labelY});
+			function unitify(unit) {
+				return function(d) {
+					return '' + d + unit;
+				};
 			}
+				
+			
+			var x = 1280;
+			var y = 1024;
+			var thumbnailMargin = 20;
+			
+			var el = window.document.querySelector('#map-container');
+			var svgimg = d3.select(el).append('svg:svg').attr('xmlns', 'http://www.w3.org/2000/svg').attr('class', 'map');
```

Comment:
What browser are you using? 



On Feb 24, 2015, 2:15 AM, at 2:15 AM, Chris <notifications@github.com> wrote:
>>  			}
>> +				
>> +			
>> +			var x = 1280;
>> +			var y = 1024;
>> +			var thumbnailMargin = 20;
>> +			
>> +			var el = window.document.querySelector('#map-container');
>> +			var svgimg = d3.select(el).append('svg:svg').attr('xmlns',
>'http://www.w3.org/2000/svg').attr('class', 'map');
>
>the only reason why there are in-line styles is that I could not get
>internal css working. I've tried your advice of embedding style
>directly under svg, but the result is the same - black nodes and
>nothing else displayed. Inkscape rendered the map correctly, but
>browser was black-nodes-only. I gave up on this, especially as nobody
>complained about the map size ;-).
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/cdaniel/wardleymapstool/pull/18/files#r25232707